src/Resources/skeleton/serializer/Encoder.php.txt

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace App\Serializer;
+
+use Symfony\Component\Serializer\Encoder\DecoderInterface;
+use Symfony\Component\Serializer\Encoder\EncoderInterface;
+
+class {{ encoder_class_name }} implements EncoderInterface, DecoderInterface
+{
+    const FORMAT = '{{ format }}';
+    
+    public function encode($data, $format, array $context = array())
+    {
+        // TODO: return your encoded data
+        return '';
+    }
+
+    public function supportsEncoding($format)
```

Comment:
Just asking: we can't do it right and add type hints (e.g. `supportsEncoding(string $format): bool`) because of the interfaces, right?

Reply:
@javiereguiluz indeed. Since there is no type-hinting in interface trying to add some in implemented method will end up with 
```
Fatal error: Declaration of YamlEncoder::supportsEncoding(string $format) must be compatible with EncoderInterface::supportsEncoding($format)
```

Reply:
In fact you could at least add the return type: https://3v4l.org/GiabR.

Reply:
@ogizanagi right! Done!